src/main/java/com/sebar/Medical/service/VisitService.java

```diff
@@ -22,32 +24,24 @@
 @RequiredArgsConstructor
 public class VisitService {
     private final VisitRepository visitRepository;
+    private final PatientRepository patientRepository;
     private final VisitMapper visitMapper;
-    private final PatientService patientService;
     private final PatientMapper patientMapper;
 
     public VisitDto addVisit(VisitCreationDto visitCreationDto) {
         Optional<Visit> visitOptional = visitRepository.findByVisitTime(visitCreationDto.getVisitTime());
-//        for (Visit visit : visitRepository.findAll()) {
-//            LocalDateTime newVisitStartTime=visitCreationDto.getVisitTime();
-//            LocalDateTime newVisitEndTime=visitCreationDto.getEndTime();
-//            LocalDateTime existingVisitStartTime=visit.getVisitTime();
-//            LocalDateTime existingVisitEndTime=visit.getEndTime();
-//
-//            if((!newVisitStartTime.isAfter(existingVisitStartTime) && newVisitStartTime.isBefore(existingVisitEndTime)) ||
-//                    (!newVisitEndTime.isAfter(existingVisitStartTime) && newVisitEndTime.isBefore(existingVisitEndTime)) ||
-//                    (!newVisitStartTime.isBefore(existingVisitStartTime) && newVisitEndTime.isAfter(existingVisitEndTime))){
-//                throw new WrongDateException("In this time slot, visit already exist");
-//            }
-//        }
+        if (visitCreationDto.getVisitTime() == null) {
+            throw new NullPointerException("Time of the visit is null");
```

Comment:
czemu null pointer exception? Jesli ktos nie przekazal odpowiedniego parametru to znaczy ze przekazal zly request (BAD_REQUEST)